src/librustc_codegen_ironox/basic_block.rs

```diff
@@ -6,5 +6,48 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
+use context::CodegenCx;
+use value::Value;
+
+macro_rules! asm {
+    ($m:expr, $($args:expr)*) => {
+        $(
+            $m.instrs.push(format!("{}\n", $args));
+        )*
+    }
+}
+
+/// The index of the parent function, and the index of the basic block
+/// in the function.
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub struct BasicBlock(pub usize, pub usize);
+
+/// A basic block.
 #[derive(Debug, PartialEq)]
-pub struct BasicBlock {}
+pub struct BasicBlockData {
+    pub label: String,
+    pub instrs: Vec<String>,
```

Comment:
Yes, at least for now.

My next PR adds `Instruction`s, so this will eventually be a `Vec<Instruction>`. I'll add documentation.

Reply:
Is there any reason you don't write directly to file. Storing up all of the text will use more RAM (but not a lot more).

Reply:
I'm changing this in my next PR (I will store `Instruction`s instead). The reason why the instructions have to be kept in memory is because they need to be preprocessed before code emission (for example, if I have time, I will have to come up with a register allocation scheme)